pkg/apiserver/authenticator/authn.go

```diff
@@ -96,12 +99,20 @@ func New(config AuthenticatorConfig) (authenticator.Request, error) {
 		authenticators = append(authenticators, serviceAccountAuth)
 	}
 
-	if len(config.KeystoneURL) > 0 {
-		keystoneAuth, err := newAuthenticatorFromKeystoneURL(config.KeystoneURL)
-		if err != nil {
-			return nil, err
+	if len(config.KeystoneConfig) > 0 {
+		if config.KeystoneAuthMode == "token" {
```

Comment:
Is it not possible to use both password and token mode at the same time in a single apiserver?